### ISSUE TITLE
Thread safety with Quantifier support

### ISSUE THREAD

**Issue by aman-goel** [SRI-CSL/yices2#276]
This relates to the current developing support for quantifiers in Yices (check pull request #259). The latest version e5314ce81282d55e889afbf51ad8bcc55986de1b is failing some travis regression tests when thread safety is enabled. Check [this travis build](https://travis-ci.org/github/SRI-CSL/yices2/builds/708471970) for more details. Failure appears in the simple [epr.smt2](https://github.com/SRI-CSL/yices2/blob/master/tests/regress/efsmt/mbqi/epr.smt2) regression test.

On trying to compile Yices master with thread safety (using ./configure --enable-thread-safety) locally, I am getting the error:
```
<path>/yices2/src/solvers/egraph/egraph.c:6242: undefined reference to `get_yices_lock'
<path>/yices2/src/solvers/egraph/egraph.c:6242: undefined reference to `release_yices_lock'
collect2: error: ld returned 1 exit status
Makefile:1051: recipe for target '../build/x86_64-pc-linux-gnu-debug/bin/yices' failed
make[2]: *** [../build/x86_64-pc-linux-gnu-debug/bin/yices] Error 1
```

Any help on resolving the above error and the issue will be very helpful. Please let me know if any detail is further needed.

Thanks.

--
Aman

**Comment by ianamason** [comment]
`get_yices_lock` is defined in `src/mt/yices_locks.[ch]` as is the other one you mention.
So you would need to link against `mt/yices_locks.o`

**Comment by BrunoDutertre** [comment]
@aman-goel The Makefile should handle this automatically. Try to do a full clean
before recompiling:

```
./configure --enable-mcsat
make all-clean
make
```

**Comment by ianamason** [comment]
I though he wanted to `--enable-thread-safety` ?

**Comment by BrunoDutertre** [comment]
Yes, Sorry

**Comment by aman-goel** [comment]
Thanks for the quick help. The issue is now resolved in 4e44a38de3a691ad1ad16948307e2e7a09ed681e.
